civihr_employee_portal/src/Service/EmergencyContactDeletionService.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Drupal\civihr_employee_portal\Service;
+
+use CRM_Hremergency_Service_EmergencyContactService as EmergencyContactService;
+
+class EmergencyContactDeletionService {
+
+  /**
+   * Deletes an emergency contact and sends a notification mail about it
+   *
+   * @param int $id
+   *   The ID of the emergency contact
+   */
+  public function delete($id) {
+    $service = new EmergencyContactService();
+    $emergencyContact = $service->find($id);
+
+    if (!$emergencyContact) {
+      $err = sprintf('Emergency contact with ID "%d" was not found', $id);
+      throw new \Exception($err);
+    }
+
+    if (!$this->canDelete($emergencyContact)) {
+      $err = 'You do not have permission to delete that emergency contact';
+      throw new \Exception($err);
+    }
+
+    $service->delete($id);
+    $this->sendNotification($emergencyContact);
+  }
+
+  /**
+   * Sends the notification mail about emergency contact deletion
+   *
+   * @param array $emergencyContact
+   */
+  protected function sendNotification($emergencyContact) {
+    if (!WebformSubmissionSettingsService::shouldSendMail()) {
```

Comment:
If I got this right, this is not really a webform submission and yet we're using the `WebformSubmissionSettingsService` here. Maybe we need a better name to this class?

Reply:
Yep, it started just as webform submissions, but it does apply to emergency contact deletion too so I'll rename it